scripts/hudson/narayana.sh

```diff
@@ -985,10 +992,10 @@ export ANT_OPTS="$ANT_OPTS $IPV6_OPTS"
 
 # run the job
 
-[ $NARAYANA_BUILD = 1 ] && build_narayana "$@"
-[ $AS_CLONE = 1 ] && clone_as "$@"
+[ $NARAYANA_BUILD = 1 ] && build_narayana "$@" # NB we should do this after building the AS since it takes longer
+[ $AS_CLONE = 1 ] && AS_BUILD=1 # keep it simple and assume that we are cloning the AS because we want to build it
+[ $AS_DOWNLOAD = 1 ] && AS_BUILD=1 # keep it simple and build it from scratch - we can revisit it later if required
 [ $AS_BUILD = 1 ] && build_as "$@"
-[ $AS_DOWNLOAD = 1 ] && download_as "$@"
```

Comment:
If you want to remove the ability to download the AS I think you should propose to remove more than this one line. But I think that should be discussed and proposed separate to this pull request. To prevent any of the current PROFILE from trying to download I think you should just change the LRA PROFILE to change AS_DOWNLOAD=1 to AS_DOWNLOAD=0 (as well as changing both AS_BUILD and AS_CLONE to 1)

Reply:
To be clear, the other things to remove would include the download_as function, as well as any reference to AS_DOWNLOAD in the script

Reply:
@tomjenkinson I have argued elsewhere in this PR about the complexity/performance trade-off and why simplification should trump generality. AS_DOWNLOAD adds no significant, if any, functional improvements, only performance, and I'd like to see the difference between downloading from team city versus building it from source. I've also argued that we can re-instate some of the complexity once we've migrated to jakarta APIs if that's what the community wants. Note that I am getting some heat from the quarkus community, who have no dependency on WildFly, about the next release of Narayana and I can't do that for them with the scripts in their current state.

Reply:
If none of the profiles use "AS_DOWNLOAD" then removing this line would have no affect. The reason it is somehow having an effect at the moment is the way you have AS_DOWNLOAD=1 in the LRA PROFILE